Mathlib/Order/SuccPred/Limit.lean

```diff
@@ -130,6 +130,12 @@ theorem not_isSuccPrelimit_succ_of_not_isMax (ha : ¬ IsMax a) : ¬ IsSuccPrelim
 theorem not_isSuccLimit_succ_of_not_isMax (ha : ¬ IsMax a) : ¬ IsSuccLimit (succ a) :=
   mt IsSuccLimit.isMax ha
 
+/-- Given `j < i` with `i` a prelimit, `IsSuccPrelimit.mid` picks an arbitrary element strictly
+between `j` and `i`. -/
+noncomputable def IsSuccPrelimit.mid {i j : α} (hi : IsSuccPrelimit i) (hj : j < i) :
```

Comment:
Is this a particularly useful definition? As the proof shows, it's basically just a direct application of `not_covBy_iff`. Maybe it could be stated in terms of `covBy` so that it doesn't need to be generalized to all the different `IsLimit` spellings?

Reply:
It's used to construct data and `Classical.indefiniteDescription` is quite long to write ... I expect this to be useful generally for transfinite induction arguments: at a limit ordinal `i`, if for each `k < i` you already constructed a family of objects over all `j < k`, you can obtain a family over all `j < i` by choosing a `k` in between for each `j < i`.

Reply:
What do you think about calling it `iooOfLT`?

Reply:
Honestly, I think the name `mid` is fine.